cfme/tests/intelligence/chargeback/test_chargeback_long_term_rates.py

```diff
@@ -0,0 +1,475 @@
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    ems = appliance.db.client['ext_management_systems']
+    rollups = appliance.db.client['metric_rollups']
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(rollups.id)
+            .join(ems, rollups.parent_ems_id == ems.id)
+            .filter(rollups.capture_interval_name == 'hourly', rollups.resource_name == vm_name,
+            ems.name == provider.name, rollups.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(rollups).filter(
+            rollups.id.in_(result.subquery())):
+        return any([record.cpu_usagemhz_rate_average,
+           record.cpu_usage_rate_average,
+           record.derived_memory_used,
+           record.net_usage_rate_average,
+           record.disk_usage_rate_average])
+    return False
+
+
+def verify_records_metrics_table(appliance, provider):
+    """Verify that rollups are present in the metric_rollups table."""
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    ems = appliance.db.client['ext_management_systems']
+    metrics = appliance.db.client['metrics']
+
+    # Capture real-time C&U data
+    ret = appliance.ssh_client.run_rails_command(
+        "\"vm = Vm.where(:ems_id => {}).where(:name => {})[0];\
+        vm.perf_capture('realtime', 1.hour.ago.utc, Time.now.utc)\""
+        .format(provider.id, repr(vm_name)))
+    assert ret.success, "Failed to capture VM C&U data:".format(ret.output)
+
+    with appliance.db.client.transaction:
+        result = (
+            appliance.db.client.session.query(metrics.id)
+            .join(ems, metrics.parent_ems_id == ems.id)
+            .filter(metrics.capture_interval_name == 'realtime',
+            metrics.resource_name == vm_name,
+            ems.name == provider.name, metrics.timestamp >= date.today())
+        )
+
+    for record in appliance.db.client.session.query(metrics).filter(
+            metrics.id.in_(result.subquery())):
+        return any([record.cpu_usagemhz_rate_average,
+            record.cpu_usage_rate_average,
+            record.derived_memory_used,
+            record.net_usage_rate_average,
+            record.disk_usage_rate_average])
+    return False
+
+
+@pytest.fixture(scope="module")
+def resource_usage(vm_ownership, appliance, provider):
+    """Retrieve resource usage values from metric_rollups table.
+
+    Chargeback reporting is done on hourly and daily rollup values and not real-time values.So, we
+    are capturing C&U data and forcing hourly rollups by running commands through
+    the Rails console.
+    """
+    average_cpu_used_in_mhz = 0
+    average_memory_used_in_mb = 0
+    average_network_io = 0
+    average_disk_io = 0
+    average_storage_used = 0
+    consumed_hours = 0
+    vm_name = provider.data['cap_and_util']['chargeback_vm']
+
+    metrics = appliance.db.client['metrics']
```

Comment:
please ask developers to create such collection in api since it is necessary for testing